frontend/src/pages/news/insideNewsPage.jsx

```diff
@@ -34,14 +37,156 @@ export default function InsideNewsPage() {
                 });
         } else {
             setArticle(newsData?.article);
-            setLikes(newsData?.likes);
         }
     }, []);
 
     useEffect(() => {
         if(article) setSections(Object.values(article?.article))
     }, [article]);
 
+    const [likeStatus, setLikeStatus] = useState(false);
+    const [initialLikeStatus, setInitialLikeStatus] = useState(false);
+    const elementRef = useRef(null);
+    const [hasRead, setHasRead] = useState(false);
+    const likeStatusRef = useRef(likeStatus);
+    const initialLikeStatusRef = useRef(initialLikeStatus);
+
+    const { user } = useContext(AuthContext)
+
+
+    useEffect(() => {
+        const observer = new IntersectionObserver(
+            ([entry]) => {
+                if (entry.isIntersecting && !hasRead) {
+                    saveInteraction('read');
+                    setHasRead(true);
+                }
+            },
+            { threshold: 0.8 }
+        );
+
+        if (elementRef.current) {
+            observer.observe(elementRef.current);
+        }
+
+        return () => {
+            if (elementRef.current) {
+                observer.unobserve(elementRef.current);
+            }
+        };
+    }, [hasRead]);
+
+    useEffect(() => {
+        const handleEvent = () => {
+            handleLikeStatus();
+        };
+
+        window.addEventListener("beforeunload", handleEvent);
+        window.addEventListener("popstate", handleEvent);
+
+        return () => {
+            window.removeEventListener("beforeunload", handleEvent);
+            window.removeEventListener("popstate", handleEvent);
+        };
+    }, []);
+
+    useEffect(() => {
+        initialLikeStatusRef.current = initialLikeStatus;
+    }, [initialLikeStatus]);
+
+    useEffect(() => {
+        likeStatusRef.current = likeStatus;
+    }, [likeStatus]);
+
+    const handleLikeStatus = async () => {
+        if (likeStatusRef.current !== initialLikeStatusRef.current) {
+            let interactionType = likeStatusRef.current ? 'like' : 'dislike';
+            saveInteraction(interactionType);
+        }
+    };
+
+    useEffect(() => {
+        const logInteraction = async () => {
+            await saveInteraction('open');
```

Comment:
You hardcode a lot of those interactionType strings. Maybe create some enum-like stuff. Though JS does not have exactly an ENUM type, but you can work it around like:

`const InteractionType = Object.freeze({`
`    LIKE: "like",`
`    DISLIKE: "dislike",`
`    COMMENT: "comment",`
`    SHARE: "share"`
});`

Something of that kind

Avoid hardcode at all cost. We have to minimize room for error in the future